gprofiler/profiler_state.py

```diff
@@ -0,0 +1,26 @@
+from threading import Event
+from typing import Union
+
+from gprofiler.utils import TemporaryDirectoryWithMode
+
+
+class ProfilerState:
+    # Class for storing generic state parameters. These parameters are the same for each profiler.
+    # Thanks to that class adding new state parameters to profilers won't result changing code in every profiler.
+    def __init__(
+        self,
+        stop_event: Event,
+        storage_dir: Union[TemporaryDirectoryWithMode, str],
```

Comment:
Why support both cases? Let's support one and be consistent across users of this class (which are either `GProfiler` or the tests code)